### ISSUE TITLE
False positive builtin-not-iterating when using map()

### ISSUE THREAD

**Issue by tiagohonorato** [PyCQA/pylint#4078]
### Steps to reproduce
Run pylint with --py3k over the following example:

```
for _ in map(str, range(5)):
	pass
```

### Current behavior
`W1638: range built-in referenced when not iterating (range-builtin-not-iterating)`

### Expected behavior

No error as `map` allows iterable.

### pylint --version output
```
$ pylint --version
pylint 2.6.0
astroid 2.4.2
Python 3.8.7 (default, Jan 20 2021, 00:00:00) 
[GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]
```
